plugins/woocommerce/includes/queue/class-wc-action-queue.php

```diff
@@ -16,52 +18,57 @@
  * A job queue using WordPress actions.
  *
  * @version 3.5.0
+ *
+ * TODO: backward compatibility
```

Comment:
Context: `woocommerce_queue_class` filter allows the specification of a replacement. We'll need an aggregate that is aware of priorities support by the queue instance and route priority flag accordingly.

Note: new interface, queue implementation, and the aggregate should be under `src`.